### PR TITLE
[simple] add failed round counter

### PR DESCRIPTION
**Pull request by igor-aptos**

### CODE REVIEW

**Review by zekun000 — COMMENTED** [review]
what's the difference between this and this https://github.com/aptos-labs/aptos-core/blob/d9d6791cfc316043e66db25da399c45ae7344539/consensus/src/round_manager.rs#L290?

**Comment by igor-aptos** [comment]
> what's the difference between this and this?
> 
> https://github.com/aptos-labs/aptos-core/blob/d9d6791cfc316043e66db25da399c45ae7344539/consensus/src/round_manager.rs#L290

@zekun000 Not sure, but some concerns:
- Are we guaranteed to not skip rounds in process_new_round_event? (i.e. to go through all?)
- how is timeout certificate handled?

**Comment by zekun000** [comment]
- we do skip rounds (jump rounds with certificate)
- not sure what do you mean about the timeout certificate handling, we move to TC.round + 1 like the QC

**Comment by igor-aptos** [comment]
Given that this will not count all successful/failed rounds, I still think it is useful to have this?

**Review by zekun000 — APPROVED** [review]

**Review by JoshLind — APPROVED** [review]

File: consensus/src/block_storage/block_store.rs [hunk 1]
```diff
@@ -56,6 +56,11 @@ pub fn update_counters_for_committed_blocks(blocks_to_commit: &[Arc<ExecutedBloc
         counters::LAST_COMMITTED_ROUND.set(block.round() as i64);
         counters::LAST_COMMITTED_VERSION.set(block.compute_result().num_leaves() as i64);
 
+        let failed_rounds = block.block().block_data().failed_authors().map(|v| v.len()).unwrap_or(0);
```
Comment (sitalkedia) [inline consensus/src/block_storage/block_store.rs:59]: Can you clarify how is it different from `aptos_consensus_timeout_rounds_count`?

Comment (igor-aptos) [reply consensus/src/block_storage/block_store.rs:59]: that one skips rounds based on how we receive certificates. you can see the discussion above

**Review by sitalkedia — APPROVED** [review]

**Pull request merged**